### ISSUE TITLE
_overlapped import error (invalid argument)

### ISSUE THREAD

**Issue by KLRussell** [pyinstaller/pyinstaller#4964]
I compiled an .exe with the code below using pyinstaller and keep getting the error in the screenshot. Normally this would trigger whenever you import asyncio or exchangelib modules because they eventually reference to the _overlapped module. Normally, running the script below in python works great, but something about running in a frozen state even with including it as data still makes this not work.

`import _overlapped
print('hi')`

![image](https://user-images.githubusercontent.com/47677315/86400479-ae05cd00-bc76-11ea-963a-915de284cbe2.png)

**Comment by bwoodsend** [comment]
It runs fine on my Windows. Can we have the full run-time traceback so we know what actually raised the OSError?

**Comment by KLRussell** [comment]
> 
> 
> It runs fine on my Windows. Can we have the full run-time traceback so we know what actually raised the OSError?

I have tried a traceback by using this code, but i am still getting the same results.

from traceback import format_exc

```
try:
    import _overlapped

    print('hi')
except:
    print(format_exc())
```

![image](https://user-images.githubusercontent.com/47677315/88093455-063a3b80-cb60-11ea-807f-edc534e09957.png)

**Comment by bwoodsend** [comment]
@KLRussell Ahh, you misunderstand what I mean by traceback. I mean the entire error message - not just those few lines you've screenshot. If you're still not sure what I mean, run the following from `cmd`:

```shell
.\test.exe >errors.txt 2>&1
```
This should create a file called `errors.txt`. Please send us that file or copy its contents to here.

**Comment by KLRussell** [comment]
> 
> 
> @KLRussell Ahh, you misunderstand what I mean by traceback. I mean the entire error message - not just those few lines you've screenshot. If you're still not sure what I mean, run the following from `cmd`:
> 
> ```shell
> .\test.exe >errors.txt 2>&1
> ```
> 
> This should create a file called `errors.txt`. Please send us that file or copy its contents to here.

First off thank you for taking the time to help troubleshoot this. Secondly, I just compiled an .exe with this code and ran a full traceback of the .exe using the method you mentioned. this is the results:

```
import _overlapped

print('hi')
```

[errors.txt](https://github.com/pyinstaller/pyinstaller/files/4955932/errors.txt)

**Comment by bwoodsend** [comment]
Sorry, I thought I replied to this yesterday. @KLRussell That is the traceback I was asking for - thanks. Unfortunately I have absolutely no idea how to fix it. @Legorooj I'm needing help with this one.

**Comment by Legorooj** [comment]
I'll have a look at this. Give me a few days.

**Comment by kjoyce77** [comment]
I am also having this issue with an identical traceback.  I have tried Python 3.8.3 with pyinstaller 3.6 as well as Python 3.7.4 with pyinstaller 3.5.

I am building on Window Version 1703 Enterprise N.  It works fine on my 1903 Home machine.  I copied the dist directory from my 1903 machine to the 1703 machine and I still get the error.  The corporate machine is virtual so that could be another possible cause of the issue.

@KLRussell what Windows version / setup do you have?

**Comment by KLRussell** [comment]
@kjoyce77, interesting. I too run Windows 10 Enterprise, Version 1709, OS Build 16299.547.

**Comment by Legorooj** [comment]
Well. I've never come across this error code: https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--1000-1299-#ERROR_NOTIFY_ENUM_DIR

Seeing as this works fine for me, frozen or not, this looks to be a harder problem to solve. @KLRussell can you please run `python -v test.py > unfrozen_imports.txt`, then build the executable with `-d imports`, and post the output of that as text?

**Comment by kjoyce77** [comment]
OK, so I discovered something new.  Being in an experimental stage I started using the default network share without thinking.  Realizing this was slowing a lot of things down I moved things to the attached storage.  It works from the attached storage.  So this could possibly be network storage protocol related or slow i/o related.

@KLRussell Are you on a network share?  If not do you have noticeably slow i/o to your storage?

**Comment by KLRussell** [comment]
@kjoyce77, yes I am running it on network share. After testing it on non-network share, you do not get this issue at all. However, this does seem to persist if ran on network share, which is odd because Non import _overlapped compiled exe's dont do this on network share. I don't mind running it on my machine, but it would be nice if it could be fixed for network share for multiple users.

**Comment by heoga** [comment]
The network share is a curious aspect.  I created a few quick shares and had difficulty reproducing but admittedly that was on a 1903 Home machine.  I have a few questions for @KLRussell and @kjoyce77 

- Are you using the one-file or one-folder bundling methods?
- Are the shares from Windows machines or Linux ones?  i.e. is Samba in play here?
- What exactly is on the network share?  Build location, compliled exes, working directory or all of the above?
- Given we're on a network share in what I'm guessing is a corporate setting, do you have security tools in operation that might be scanning exes for malicious, or unapproved software?  My experience is these can often get quite aggressive and perform poorly on network locations.

**Comment by kjoyce77** [comment]
@heoga In my case here are my answers.
- I have done both one-file and one-folder with the same results
- I am told the shares are NetApp.  My contact did say CIFS was being used but he also gave me the name of a subject-matter expert who might have more details.
- All of the above are on the share
- There are definitely security tools and it is very possible this is not something we are "supposed" to do.  It essentially isn't an issue for me anymore since I will be documenting for all users that they need to extract the folder to their C: drive.  Still, it would be nice to figure out if we could at least trigger a warning for this situation..
